sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -144,6 +147,35 @@ case class Join(
     }
   }
 
+  def outerProjectList: Seq[NamedExpression] = {
```

Comment:
did you mean "column" instead of row? columns of the same name should only appear once.

also this code should go into the analyzer.